src/components/core/GeoChart.tsx

```diff
@@ -40,28 +52,49 @@ export type IGeoChartInnerProps = ICoreGeoChartProps &
     IDataSourceProviderInjectedProps &
     WrappedComponentProps;
 
+export interface IGeoChartInnerState {
+    legendIndexEnabled: boolean[];
+    legendNamesEnabled: string[];
+}
+export interface IGeoChartInnerOptions {
+    geoData: IGeoData;
+    series: any;
+    sizeValues: number[];
+}
 /**
  * Geo Chart react component
  */
-export class GeoChartInner extends BaseVisualization<IGeoChartInnerProps, {}> {
+export class GeoChartInner extends BaseVisualization<IGeoChartInnerProps, IGeoChartInnerState> {
     public static defaultProps: Partial<IGeoChartInnerProps> = {
         ...commonDefaultProps,
         chartRenderer: renderChart,
         legendRenderer: renderLegend,
-        legendPostion: TOP,
+        legendPosition: TOP,
     };
+    private geoChartOptions: IGeoChartInnerOptions;
+    constructor(props: IGeoChartInnerProps) {
+        super(props);
+        this.state = {
+            legendIndexEnabled: [],
+            legendNamesEnabled: [],
+        };
+        this.onLegendItemClick = this.onLegendItemClick.bind(this);
```

Comment:
use arrow function instead of binding `this`

Reply:
fixed